modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1536,6 +1583,355 @@ MULC_SIMD(float, float)
+divc_simd_common_impl(scale_tag_t s_tag, const SRC* inx, DST* outx,
+                      const v_float32& v_scalar, const v_float32& v_scale,
+                      const int length)
+{
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+
+    v_float32 v_zero = vx_setzero_f32();
+    v_float32 v_mask = (v_scalar == v_zero);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = vg_load_f32(inx);
+            v_float32 a2 = vg_load_f32(&inx[nlanes/2]);
+
+            v_store_i16(outx, v_round(v_select(v_mask, v_zero, div_op(s_tag, a1, v_scalar, v_scale))),
+                              v_round(v_select(v_mask, v_zero, div_op(s_tag, a2, v_scalar, v_scale))));
+        }
+
+        if (x < length)
+        {
+            x = length - nlanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC>
+CV_ALWAYS_INLINE int divc_simd_common_impl(scale_tag_t s_tag, const SRC* inx,
+                                           uchar* outx, const v_float32& v_scalar,
+                                           const v_float32& v_scale, const int length)
+{
+    constexpr int nlanes = v_uint8::nlanes;
+
+    v_float32 v_zero = vx_setzero_f32();
+    v_float32 v_mask = (v_scalar == v_zero);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = vg_load_f32(inx);
+            v_float32 a2 = vg_load_f32(&inx[nlanes/4]);
+            v_float32 a3 = vg_load_f32(&inx[nlanes/2]);
+            v_float32 a4 = vg_load_f32(&inx[3 * nlanes/4]);
+
+            vx_store(outx, v_pack_u(v_pack(v_round(v_select(v_mask, v_zero, div_op(s_tag, a1, v_scalar, v_scale))),
+                                           v_round(v_select(v_mask, v_zero, div_op(s_tag, a2, v_scalar, v_scale)))),
+                                    v_pack(v_round(v_select(v_mask, v_zero, div_op(s_tag, a3, v_scalar, v_scale))),
+                                           v_round(v_select(v_mask, v_zero, div_op(s_tag, a4, v_scalar, v_scale))))));
+        }
+
+        if (x < length)
+        {
+            x = length - nlanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC>
+CV_ALWAYS_INLINE int divc_simd_common_impl(scale_tag_t s_tag, const SRC* in,
+                                           float* out, const v_float32& v_scalar,
+                                           const v_float32& v_scale, const int length)
+{
+    constexpr int nlanes = v_float32::nlanes;
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
```

Comment:
let's assume `length < nlanes` is true
that we go to the next cycle below

`x = 0` and yes `(if x < length)` is true, then x become  negative because `x = length - nlanes`
Then we go to the first loop again and condition `x <= length-nlanes` become true and cycle set off it first step. In this case `in[x]` is out-of-bound memory

Reply:
Yes, you are right. So for this case there is check a level higher on the call stack. Please take a look [here](https://github.com/opencv/opencv/pull/21474/files#diff-0a90dc2fdb3d4d8978392dc6818579e0c3a47db58e3ced9211522d8dd8158cb3R1702)